finance/management/commands/xformnetfilerawdata.py

```diff
@@ -0,0 +1,365 @@
+
+                # of=official, pf=primarily, ic=independent
+                benefactor.name = row['tran_NamL'].strip()
+                benefactor.type = benefactor.type or 'PF'  # TODO: fix
+                benefactor.city = bf_city
+                benefactor.state = bf_state
+                benefactor.zip_code = bf_zip_code
+                benefactor.locality = bf_city
+                benefactor.save()
+
+            elif row['entity_Cd'] == 'PTY':  # political party
+                raise NotImplementedError("Form 460 Sched. A contributions "
+                                          "from political parties.")
+
+            else:
+                raise ValueError("Entity type not expected: %s" % (
+                    row['entity_Cd']))
+
+            return benefactor, bf_zip_code
+        benefactor, bf_zip_code = parse_benefactor(row)
+
+        def parse_form_and_report_period(row):
+            # Create/get the relevant form objects.
+            f460A, _ = models.Form.objects.get_or_create(  # noqa
+                name='460 Schedule A', text_id='460A',
+                submission_frequency='SA')
+
+            report_date = date_parse(row['tran_Date'])
+            reporting_period, _ = models.ReportingPeriod.objects.get_or_create(
+                form=f460A,
+                period_start=datetime.datetime(report_date.year, 1, 1),
+                period_end=datetime.datetime(report_date.year, 12, 31))
+
+            return reporting_period
+        reporting_period = parse_form_and_report_period(row)  # noqa
+
+        def parse_beneficiary(row):
+            # Parse and save the beneficiary, contribution.
+            agency_shortcut = row['filerId'].split('-')[0]
+            agency = self.get_agency(agency_shortcut)
+            if agency:
+                state, _ = State.objects.get_or_create(short_name='CA')
+                if state.name is None or state.name == '':
+                    state.name = 'California'
+                    state.save()
+                locality, _ = City.objects.get_or_create(
+                    short_name=agency['shortcut'], name=agency['name'],
+                    state=state)
+            else:
+                locality = None
+
+            beneficiary = self.get_committee_beneficiary(row)
+            beneficiary.locality = locality
+            beneficiary.name = row['filerName'].strip()
+            beneficiary.type = 'PF'  # ok
+            # beneficiary.address = '?'  # TODO: fix
+            beneficiary.save()
+            return beneficiary
+        beneficiary = parse_beneficiary(row)
+
+        def parse_candidate_and_office(row):
+            import re
+            res = (
+                # David Alvarez for Mayor 2014
+                '^(?P<name>.*?)\s+for\s+(?P<office>.*?)(?P<year>[0-9]+).*$',
+                # Scott Sanborn City Council 2016
+                '^(?P<name>.*? .*?)\s+(?P<office>.*?)(?P<year>[0-9]+).*$',
+            )
```

Comment:
There was one candidate for Mayor in Oakland who did not have her full name in the committee name.  I don't think we can count on all committee names to follow a known format.  On the other hand most will so maybe we just need to keep an eye out for those that do not.

> On Jan 13, 2016, at 6:52 PM, Tom Dooner <notifications@github.com> wrote:
> 
> In finance/management/commands/xformnetfilerawdata.py:
> 
> > +            beneficiary.locality = locality
> > +            beneficiary.name = row['filerName'].strip()
> > +            beneficiary.type = 'PF'  # ok
> > +            # beneficiary.address = '?'  # TODO: fix
> > +            beneficiary.save()
> > +            return beneficiary
> > +        beneficiary = parse_beneficiary(row)
> > +
> > +        def parse_candidate_and_office(row):
> > +            import re
> > +            res = (
> > +                # David Alvarez for Mayor 2014
> > +                '^(?P<name>.*?)\s+for\s+(?P<office>.*?)(?P<year>[0-9]+).*$',
> > +                # Scott Sanborn City Council 2016
> > +                '^(?P<name>.*? .*?)\s+(?P<office>.*?)(?P<year>[0-9]+).*$',
> > +            )
> Does this actually get good results?
> 
> —
> Reply to this email directly or view it on GitHub.
>